quodlibet/quodlibet/qltk/songlistcolumns.py

```diff
@@ -279,14 +279,23 @@ def _apply_value(self, model, iter_, cell, stamp):
             except (OverflowError, ValueError, OSError):
                 text = u""
             else:
-                today = datetime.datetime.now().date()
-                days = (today - date).days
-                if days == 0:
-                    format_ = "%X"
-                elif days < 7:
-                    format_ = "%A"
+                format_setting = config.gettext("settings",
+                                      "datecolumn_timestamp_format")
+
+                # use default behaviour-format
+                if not format_setting:
```

Comment:
I suppose it could be swapped. I started with `== ""` and changed it to `not s` in an attempt to fix some failed check (mentioned earlier), and at that point the new order maybe makes more sense. 

I can try to look into how tests work to add some.